Add saturation metrics for network and memory

luhenry opened the pull request:
### What does this PR do?

Add saturation metrics for network and memory.

### Motivation

We currently gather utilization and error metrics of network (bytes received and sent, number of packets) but not the saturation metrics (number of dropped packets). Similarly for memory, we gather the utilization metrics (free, used) but not the saturation (page in/out, swap in/out).

Having this additional data is needed to apply the USE Method [1] for system performance analysis.

[1] https://www.brendangregg.com/usemethod.html

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

After installing the agent, you should have the following additional metrics on the given platforms:
| metrics                                                 | platforms                    |
| ------------------------------------------------------- | ---------------------------- |
| system.net.packets_in.drop, system.net.packets_out.drop | all platforms                |
| system.io.block_in, system.io.block_out                 | all platforms except windows |
| system.swap.swap_in, system.swap.swap_out               | all platforms except windows |

### Checklist
<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

luhenry committed:
Add saturation metrics for network and memory

We currently gather utilization and error metrics of network (bytes received and sent, number of packets) but not the saturation metrics (number of dropped packets). Similarly for memory, we gather the utilization metrics (free, used) but not the saturation (page in/out, swap in/out).

Having this additional data is needed to apply the USE Method [1] for system performance analysis.

[1] https://www.brendangregg.com/usemethod.html

mx-psi commented on pkg/collector/corechecks/net/network.go:
```diff
@@ -199,8 +199,10 @@ func submitInterfaceMetrics(sender aggregator.Sender, interfaceIO net.IOCounters
 	sender.Rate("system.net.bytes_rcvd", float64(interfaceIO.BytesRecv), "", tags)
 	sender.Rate("system.net.bytes_sent", float64(interfaceIO.BytesSent), "", tags)
 	sender.Rate("system.net.packets_in.count", float64(interfaceIO.PacketsRecv), "", tags)
+	sender.Rate("system.net.packets_in.drop", float64(interfaceIO.Dropin), "", tags)
```
This will only be used when the Python version of the check is not available (e.g. on IoT or Android environments). You probably want to open a PR on integrations-core for the Python check

luhenry replied:
I've submitted https://github.com/DataDog/integrations-core/pull/10551 which should IIUC complete this PR.

mx-psi commented:
@luhenry this needs [a release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) and the CLA passing for it to be mergeable

luhenry committed:
Add release notes

luhenry committed:
Merge branch 'main' of https://github.com/DataDog/datadog-agent into luhenry/saturation-metrics

luhenry committed:
Try memory saturation metrics as Rate rather than Gauge

luhenry commented:
I'm not sure we want to have the memory saturation metrics as Gauge rather than Rate. Experimenting with that and possibly reverting afterwards.

luhenry committed:
Merge branch 'main' of https://github.com/DataDog/datadog-agent into luhenry/saturation-metrics

luhenry committed:
Rename system.swap.page_{in,out} to system.io.block_{in,out}

From further research and looking deeper in the Linux kernel, the pgpgin and pgpgout values in /proc/vmstat are updated on every read and write on a block device [1]. From also verifying in the sources of vmstat [2], these values are used to report bi and bo.

[1] https://github.com/torvalds/linux/blob/2b14864acbaaf03d9c01982e243a84632524c3ac/block/blk-core.c#L989-L994
[2] https://gitlab.com/procps-ng/procps/-/blob/b2c4fcfddbb12ea3820106413df50c24732cc8d2/vmstat.c#L430-431

luhenry committed:
Move system.io.block_{in,out} to disk/iostats

luhenry committed:
Merge branch 'main' of https://github.com/DataDog/datadog-agent into luhenry/saturation-metrics
